src/credentials/CHIPCert.cpp

```diff
@@ -695,6 +698,144 @@ CHIP_ERROR ChipDN::GetCertFabricId(uint64_t & fabricId) const
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ChipDN::DecodeFromTLV(TLVReader & reader)
+{
+    CHIP_ERROR err;
+    TLVType outerContainer;
+    Tag tlvTag;
+    uint32_t tlvTagNum;
+    OID attrOID;
+    bool attrIsPrintableString;
```

Comment:
Is there a reason to declare all these up top instead of where used?

Reply:
updated accordingly